ReactWindows/ReactNative/Views/Web/ReactWebViewManager.cs

```diff
@@ -0,0 +1,306 @@
+
+                    var body = source.Value<string>("body");
+                    if (body != null)
+                    {
+                        request.Content = new HttpStringContent(body);
+                    }
+
+                    view.NavigateWithHttpRequestMessage(request);
+                    return;
+                }
+            }
+            
+            view.Navigate(new Uri(BLANK_URL));
+        }
+
+        /// <summary>
+        /// Receive events/commands directly from JavaScript through the 
+        /// <see cref="UIManagerModule"/>.
+        /// </summary>
+        /// <param name="view">
+        /// The view instance that should receive the command.
+        /// </param>
+        /// <param name="commandId">Identifer for the command.</param>
+        /// <param name="args">Optional arguments for the command.</param>
+        public override void ReceiveCommand(WebView view, int commandId, JArray args)
+        {
+            switch (commandId)
+            {
+                case CommandGoBack:
+                    if (view.CanGoBack) view.GoBack();
+                    break;
+                case CommandGoForward:
+                    if (view.CanGoForward) view.GoForward();
+                    break;
+                case CommandReload:
+                    view.Refresh();
+                    break;
+                default:
+                    throw new InvalidOperationException(
+                        $"Unsupported command '{commandId}' received by '{typeof(ReactWebViewManager)}'.");
+            }
+        }
+
+        /// <summary>
+        /// Called when view is detached from view hierarchy and allows for 
+        /// additional cleanup by the <see cref="ReactWebViewManager"/>.
+        /// </summary>
+        /// <param name="reactContext">The react context.</param>
+        /// <param name="view">The view.</param>
+        public override void OnDropViewInstance(ThemedReactContext reactContext, WebView view)
+        {
+            view.NavigationCompleted -= OnNavigationCompleted;
+            view.NavigationStarting -= OnNavigationStarting;
+        }
+
+        /// <summary>
+        /// Creates a new view instance of type <see cref="WebView"/>.
+        /// </summary>
+        /// <param name="reactContext">The react context.</param>
+        /// <returns>The view instance.</returns>
+        protected override WebView CreateViewInstance(ThemedReactContext reactContext)
+        {
+            return new WebView();
+        }
+
+        /// <summary>
+        /// Subclasses can override this method to install custom event 
+        /// emitters on the given view.
+        /// </summary>
+        /// <param name="reactContext">The react context.</param>
+        /// <param name="view">The view instance.</param>
+        protected override void AddEventEmitters(ThemedReactContext reactContext, WebView view)
+        {
+            view.NavigationCompleted += OnNavigationCompleted;
```

Comment:
Failure event is deprecated with Windows 10 OnCompleted should be used instead of it and check e.IsSuccess.